Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -351,20 +348,18 @@ extension ProductStore {
         storage.saveIfNeeded()
     }
 
-    /// Deletes any Storage.Product with the specified `siteID`
+    /// Updates (OR Inserts) the specified ReadOnly Product Entities *in a background thread*.
+    /// Also deletes existing products if requested.
+    /// `onCompletion` will be called on the main thread!
     ///
-    func deleteStoredProducts(siteID: Int64) {
-        let storage = storageManager.viewStorage
-        storage.deleteProducts(siteID: siteID)
-        storage.saveIfNeeded()
-    }
-
-    /// Updates (OR Inserts) the specified ReadOnly Product Entities *in a background thread*. onCompletion will be called
-    /// on the main thread!
-    ///
-    func upsertStoredProductsInBackground(readOnlyProducts: [Networking.Product], onCompletion: @escaping () -> Void) {
+    func upsertStoredProductsInBackground(readOnlyProducts: [Networking.Product],
+                                          shouldDeleteExistingProducts: Bool = false,
+                                          onCompletion: @escaping () -> Void) {
         let derivedStorage = sharedDerivedStorage
         derivedStorage.perform {
+            if shouldDeleteExistingProducts, let id = readOnlyProducts.first?.siteID {
```

Comment:
nit: maybe we can DI the site ID to the function?

Reply:
Updated in e41ed55e66e4b175a3c736195205db84835b9413 👍